site/content/en/docs/Guides/feature-stages.md

```diff
@@ -0,0 +1,131 @@
+  If you have multiple clusters that can be upgraded independently, you may be able to relax this restriction.
+
+> Note: Please do try `Beta` features and give feedback on them! After they exit beta, it may not be practical for us
+> to make more changes.
+
+### Stable
+
+A `Stable` feature means:
+
+* The feature is enabled and the corresponding feature gate no longer exists.
+* Stable versions of features will appear in released software for many subsequent versions.
+
+## Feature Stage Indicators
+
+There are a variety of features with Agones, how can we determine what stage each feature is in?
+
+Below are indicators for each type of functionality that can be used to determine the feature stage for a given aspect
+of Agones.
+
+### Custom Resource Definitions (CRDs)
+
+This refers to Kubernetes resource for Agones, such as `GameServer`, `Fleet` and `GameServerAllocation`.
+
+#### New CRDs
+
+For new resources, the stage of the resource will be indicated by the `apiVersion` of the resource.
+
+For example: `apiVersion: "agones.dev/v1"` is a `stable` resource, `apiVersion: "agones.dev/v1beta1"` is a `beta`
+ stage resource, and `apiVersion: "agones.dev/v1alpha1"` is an `alpha` stage resource.
+
+#### New CRD attributes
+
+`alpha` and `beta` attributes will have a corresponding `alpha` or `beta` parent element in their configuration to
+delineate their feature stage.
+
+If no such parent exists, this attribute is a `stable` feature.
+
+For example, if we were to add a hypothetical `alpha` configuration option of `timeoutSeconds` to the `GameServer`
+ resource, it would be configured like so:
+ 
+```yaml
+apiVersion: "agones.dev/v1"
+kind: GameServer
+metadata:
+  generateName: "simple-udp-"
+spec:
+  alpha: # this is the alpha feature block
+    timeoutSeconds: 30
+  ports:
+  - name: default
+    portPolicy: Dynamic
+    containerPort: 7654
+  template:
+    spec:
+      containers:
+      - name: simple-udp
+        image: gcr.io/agones-images/udp-server:0.15
+``` 
+
+As resource attributes progress through their stages, there may be breaking changes, as backward conversion between
+ `alpha`, `beta` and `stable` positioning in the resource configuration may not be guaranteed.
+
+### Agones Game Server SDK
+
+Any `alpha` or `beta` Game Server SDK functionality will be a subpackage of the `sdk` package. For example
+, functionality found in a `sdk.alphav1` package should be considered at the `alpha` feature stage. 
+
+Any other SDK functionality not marked as `alpha` or `beta` can be assumed to be `stable`.
+
+### REST & gRPC APIs 
+
+REST and gRPC API will have versioned paths were appropriate to indicate their feature stage.
+
+For example, a REST API with a prefix of `v1alpha1` is an `alpha` stage feature: 
+`http://api.example.com/v1alpha1/exampleaction`.
+
+Similar to the SDK, Any `alpha` or `beta` gRPC functionality will be a subpackage of the main API package.
```

Comment:
How about if there is a breaking change to the v1 API?

Reply:
This is a good question. Ideally, we never want this to happen - since the SDK is so important -- but I may be being optimistic that it won't ever happen - or at least to the degree that we want a whole new implementation of the SDK.

I'm almost tempted to say "handle it if/when we need to?, and hopefully we won't have to?"

Reply:
Oh wait, wrong section - this isn't the SDK section, this is the API section.

So my only concern here, is that if we do a v1/v2 type API surface, it's completely different from the SDK. Maybe that is fine though? I'd like to have consistency however.

WDYT?

Reply:
Optimistically, I also expect major versioning would not happen However, it wouldn't hurt to mention that if there is any required incompatible change to the stable version there should be an extensive communication and planning.

Reply:
This sounds like a great addition as well :+1:

Reply:
How strongly do you feed about having a v1/v2 interface/package for the gRPC/REST endpoints.

I'm thinking I might be backtracking on my position of not doing this - just because it's going to be really hard to go back and slot this in if we need it (SDK comes to mind).

I'm not a huge fan of packages called `v1` (hello K8s), but I can't think of a better way.

Reply:
Version in the package name for gRPC for v1 is optional. 
For REST, I have always seen APIs to be versioned, but couldn't find a recommendation specific to v1 versioning. I would recommend designing REST API with version in the path to avoid URL changes between versions.